### PR TITLE
Dlite plugin for singlefiledatanode

### PR DESCRIPTION
**Pull request by francescalb**
Added singlefileconverter function
    
    This is a function used by oteapi and dlite and includes
    a dlite plugin for a singlefileconverter.
    
    Note that since options are noe allowed in the from_bytes
    class method of dlite plugins the converter now makes a hack
    with a temporary file and then uses the load.

### CODE REVIEW

File: dlite_plugins/singlefiledatanode.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+class singlefiledatanode(dlite.DLiteStorageBase):
+    """General description of the Python storage plugin."""
+
+    def open(self, location, options=None):
+        """Open storage.
+
+        Arguments:
+            location:
+            options: Additional options for this storage driver.
```
Comment (jesper-friis) [inline dlite_plugins/singlefiledatanode.py:23]: Arguments could be better documented. 
Especially the `driver` option would be good to have documented.

File: execflow/data/setup_dlite.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+from pathlib import Path
+
+import dlite
+
+
+def setup_dlite():
+    """Setup DLite search paths for datamodels and storage plugins."""
+    pkgdir = Path(__file__).resolve().parent.parent.parent
+    dlite.python_storage_plugin_path.append(pkgdir / "dlite_plugins")
```
Comment (jesper-friis) [inline execflow/data/setup_dlite.py:13]: Shouldn't the search path for datamodels also be setup in this function?

    dlite.storage_path.append(...)

**Review by jesper-friis — COMMENTED** [review]
Looks nice. Added two comments

File: execflow/data/setup_dlite.py (hunk 2 above)
Comment (francescalb) [reply execflow/data/setup_dlite.py:13]: why, if we do not need it?

File: dlite_plugins/singlefiledatanode.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    def load(self, id=None) -> dlite.Instance:  # noqa: ARG002
+        """
+        Load a state file and return it as a DLite instance
+
+        Returns:
+            A DLite Instance corresponding to the given filename.
+
+        """
+        return dlite.Instance.from_location(self.options.driver, self.location)
```
Comment (jesper-friis) [inline dlite_plugins/singlefiledatanode.py:40]:
```suggestion
        return dlite.Instance.from_location(self.driver, self.location, options=self.options)
```

**Review by jesper-friis — COMMENTED** [review]
Added a suggestion for passing options on to the underlying driver

**Review by jesper-friis — APPROVED** [review]

**Pull request merged**